applications/luci-app-simple-adblock/htdocs/luci-static/resources/simple-adblock/status.js

```diff
@@ -0,0 +1,306 @@
+// Copyright 2022 Stan Grishin <stangri@melmac.ca>
+// This code wouldn't have been possible without help from [@vsviridov](https://github.com/vsviridov)
+
+"require ui";
+"require rpc";
+"require form";
+"require baseclass";
+
+var pkg = {
+	get Name() { return 'simple-adblock'; },
+	get URL() { return 'https://docs.openwrt.melmac.net/' + pkg.Name + '/'; },
+};
+
+var getInitList = rpc.declare({
+	object: "luci." + pkg.Name,
+	method: "getInitList",
+	params: ["name"],
+});
+
+var getInitStatus = rpc.declare({
+	object: "luci." + pkg.Name,
+	method: "getInitStatus",
+	params: ["name"],
+});
+
+var getPlatformSupport = rpc.declare({
+	object: "luci." + pkg.Name,
+	method: "getPlatformSupport",
+	params: ["name"],
+});
+
+var _setInitAction = rpc.declare({
+	object: "luci." + pkg.Name,
+	method: "setInitAction",
+	params: ["name", "action"],
+	expect: { result: false },
+});
+
+var RPC = {
+	listeners: [],
+	on: function on(event, callback) {
+		var pair = { event: event, callback: callback }
+		this.listeners.push(pair);
+		return function unsubscribe() {
+			this.listeners = this.listeners.filter(function (listener) {
+				return listener !== pair;
+			});
+		}.bind(this);
+	},
+	emit: function emit(event, data) {
+		this.listeners.forEach(function (listener) {
+			if (listener.event === event) {
+				listener.callback(data);
+			}
+		});
+	},
+	getInitList: function getInitList(name) {
+		getInitList(name).then(function (result) {
+			this.emit('getInitList', result);
+		}.bind(this));
+	},
+	getInitStatus: function getInitStatus(name) {
+		getInitStatus(name).then(function (result) {
+			this.emit('getInitStatus', result);
+		}.bind(this));
+	},
+	getPlatformSupport: function getPlatformSupport(name) {
+		getPlatformSupport(name).then(function (result) {
+			this.emit('getPlatformSupport', result);
+		}.bind(this));
+	},
+	setInitAction: function setInitAction(name, action) {
+		_setInitAction(name, action).then(function (result) {
+			this.emit('setInitAction', result);
+		}.bind(this));
+	},
+}
+
+var status = baseclass.extend({
+	render: function () {
+		return Promise.all([
+			L.resolveDefault(getInitStatus(), {}),
+			//			L.resolveDefault(getGateways(), {}),
+		]).then(function (data) {
+			var replyStatus = data[0];
```

Comment:
```suggestion
			var replyStatus = data[0];
			var status = replyStatus[pkg.Name];
```

and then everywhere below where `replyStatus[pkg.Name]` is used it can be simplified to just `status`... There's like ~25 places where this can improve readability.

Reply:
Thank you, fixed!